openedx_ledger/utils.py

```diff
@@ -46,3 +49,26 @@ def create_idempotency_key_for_transaction(ledger, quantity, is_initial_deposit=
         }
     hashed_metadata = hashlib.md5(str(idpk_data).encode()).hexdigest()
     return f'{ledger.idempotency_key}-{quantity}-{hashed_metadata}'
+
+
+def calculate_transactions_balance(input_transactions_queryset):
+    """
+    Calculate the balance of the given transactions, taking reversals into account.
+
+    WARNING: All transactions should be part of the same ledger, or at least the same enterprise customer, or else the
+    output probably wouldn't be useful for anything.
```

Comment:
small, non-blocking idea: you could enforce this via the method signature by requiring a `ledger` argument and then tacking a `.filter(ledger=ledger)` onto your `transactions` expression below.

Reply:
or alternatively, make this a method of the `Ledger` model and add a `.filter(ledger=self)` to the queryset provided as input.

Reply:
yea, that sounds like a good idea to keep it part of the ledger model.

Reply:
Updated to add just a `transactions_filter` dict parameter to the Ledger.balance() function.